src/main/frontend/fs/watcher_handler.cljs

```diff
@@ -129,6 +132,55 @@
       ;; return nil, otherwise the entire db will be transferred by ipc
       nil)))
 
+(defn preload-graph-homepage-files!
+  "Preload the homepage file for the current graph.
+
+   Prerequisites:
+   - current graph is set
+   - config is loaded"
+  []
+  (when-let [repo (state/get-current-repo)]
+    (when (and (not (state/loading-files? repo))
+               (config/local-db? repo))
+      (let [repo-dir (config/get-repo-dir repo)
+            page-name (if (state/enable-journals? repo)
+                        (date/today)
+                        (or (:page (state/get-default-home)) "Contents"))
+            page-name (util/page-name-sanity-lc page-name)
+            file-rpath (or (:file/path (db/get-page-file page-name))
+                           (let [format (state/get-preferred-format repo)
+                                 ext (config/get-file-extension format)
+                                 file-name (if (state/enable-journals? repo)
+                                             (date/journal-title->default (date/today))
+                                             (or (:page (state/get-default-home)) "contents"))
+                                 parent-dir (if (state/enable-journals? repo)
+                                              (config/get-journals-directory)
+                                              (config/get-pages-directory))]
+                             (str parent-dir "/" file-name "." ext)))]
+        (prn ::preload-homepage repo-dir file-rpath)
+        (p/let [file-exists? (fs/file-exists? repo-dir file-rpath)
+                _ (when file-exists?
+                    ;; BUG: avoid active-editing block content overwrites incoming fs changes
```

Comment:
If an empty page is in editing mode(the first block). After fs watcher event fires(new contents from disk), the first block is always overwriten by an empty editing block.

It seems the bug is react-related.